tests/end2end/e2e_sklearn.py

```diff
@@ -0,0 +1,350 @@
+    create_component,
+    add_dataset_to_node,
+    start_nodes,
+    kill_subprocesses,
+    clear_node_data,
+    clear_experiment_data,
+    clear_researcher_data,
+    create_researcher,
+    get_data_folder,
+    create_multiple_nodes
+)
+
+from experiments.training_plans.sklearn import (
+    PerceptronTraining,
+    SGDRegressorTrainingPlan,
+    SGDClassifierTrainingPlan,
+    SkLearnClassifierTrainingPlanDeclearn,
+    SGDRegressorTrainingPlanDeclearn
+)
+
+from fedbiomed.researcher.federated_workflows import Experiment
+from fedbiomed.researcher.aggregators.fedavg import FedAverage
+from fedbiomed.common.metrics import MetricTypes
+
+from sklearn import datasets
+import numpy as np
+
+# Set up nodes and start
+@pytest.fixture(scope="module", autouse=True)
+def setup(port, post_session, request):
+    """Setup fixture for the module"""
+
+    with create_multiple_nodes(port, 3) as nodes:
+        node_1, node_2, node_3 = nodes
+
+        # Create researcher component
+        researcher = create_researcher(
+            port=port,
+            config_name="config_researcher_sklearn.ini",
+            config_sections={'server': {'port': port}},
+        )
+
+        # Starts the nodes
+        node_processes, thread = start_nodes([node_1, node_2, node_3])
+
+        # Generate datasets
+        p1, p2, p3 = generate_sklearn_classification_dataset()
+        dataset = {
+            "name": "MNIST",
+            "description": "MNIST DATASET",
+            "tags": "#csv-dataset-classification",
+            "data_type": "csv"}
+
+
+        d1 = {**dataset, "path": p1}
+        d2 = {**dataset, "path": p2}
+        d3 = {**dataset, "path": p3}
+        add_dataset_to_node(node_1, d1)
+        add_dataset_to_node(node_2, d2)
+        add_dataset_to_node(node_3, d3)
+
+
+        # Add MNIST dataset
+        datafolder = get_data_folder('')
+        dataset = {
+            "name": "MNIST",
+            "description": "MNIST DATASET",
+            "tags": "#MNIST,#dataset",
+            "data_type": "default",
+            "path": datafolder
+        }
+        add_dataset_to_node(node_1, dataset)
+        add_dataset_to_node(node_2, dataset)
+        add_dataset_to_node(node_3, dataset)
+
+
+        data_path = os.path.join('notebooks', 'data', 'CSV', 'pseudo_adni_mod.csv')
+        dataset = {
+            "name": "Adni dataset",
+            "description": "Adni DATASET",
+            "tags": "#adni",
+            "data_type": "csv",
+            "path": data_path}
+
+        add_dataset_to_node(node_1, dataset)
+        add_dataset_to_node(node_2, dataset)
+        add_dataset_to_node(node_3, dataset)
+
+        time.sleep(1)
+
+        # Good to wait 3 second to give time to nodes start
+        print("Sleep 5 seconds. Giving some time for nodes to start")
+        time.sleep(5)
+
+
+        # Run tests
+        yield node_1, node_2, node_3, researcher
+
+
+        kill_subprocesses(node_processes)
+        thread.join()
+
```

Comment:
Hi @ybouilla,

I added this but it is not mandatory because once all the node processes are killed the thread will finish. However, it was just to explicitly show that that is a thread running to follow node processes. I can add it to other tests as well.